gate-core/src/main/groovy/com/netflix/spinnaker/gate/config/AuthConfig.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ * Copyright 2023 Apple, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.config;
+
+import com.netflix.spinnaker.fiat.shared.FiatClientConfigurationProperties;
+import com.netflix.spinnaker.fiat.shared.FiatPermissionEvaluator;
+import com.netflix.spinnaker.fiat.shared.FiatStatus;
+import com.netflix.spinnaker.gate.filters.FiatSessionFilter;
+import com.netflix.spinnaker.gate.services.ServiceAccountFilterConfigProps;
+import com.netflix.spinnaker.kork.annotations.NonnullByDefault;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityCustomizer;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.web.authentication.AnonymousAuthenticationFilter;
+
+@Configuration
+@EnableConfigurationProperties({
+  ServiceConfiguration.class,
+  ServiceAccountFilterConfigProps.class,
+  FiatClientConfigurationProperties.class
```

Comment:
Curious why this moved from GateConfig

Reply:
IntelliJ doesn't support Spring+Groovy very well, so I moved it to a Java config to make it show up. Not really required, but it helps with development.